src/abstractdataframe/join.jl

```diff
@@ -57,8 +57,13 @@ struct DataFrameJoiner
         end
 
         for df in (dfl_on, dfr_on), col in eachcol(df)
-            if any(x -> (x isa Union{Complex, Real}) & isnan(x), col)
-                throw(ArgumentError("NaN values in key columns are not allowed"))
+            if any(x -> (x isa Union{Complex, Real}) &&
+                        (isnan(x) || real(x) === -0.0 || imag(x) === -0.0), col)
```

Comment:
```suggestion
                        (isnan(x) || isequal(real(x), -0.0) || isequal(imag(x), -0.0)), col)
```

Reply:
ah - of course. I was over influenced by your comment not to use `isequal`, but here it is the opposite and it has to be used 😄.